File: clases 25-11-2025/Tipos de clases c#/clase normal con interfaz/clases_abstractas/Estudiante.cs
public abstract class Estudiante
{
    public string Nombre { get; set; }
    public int Grado { get; set; }

    protected Estudiante(string nombre, int grado)
    {
        Nombre = nombre;
        Grado = grado;
    }

    // Constructor copia
    protected Estudiante(Estudiante otro)
    {
        Nombre = otro.Nombre;
        Grado = otro.Grado;
    }

    public abstract void Estudiar();
}

File: clases 25-11-2025/Tipos de clases GO/clases_normales/Estudiante.go
package main

type Estudiante struct {
	nombre string
	grado  int
}

func NewEstudiante(nombre string, grado int) Estudiante {
	return Estudiante{
		nombre: nombre,
		grado:  grado,
	}
}

func NewEstudianteCopia(otro Estudiante) Estudiante {
	return Estudiante{
		nombre: otro.nombre,
		grado:  otro.grado,
	}
}
